types/block.go

```diff
@@ -1157,16 +1169,23 @@ func (roots IntermediateStateRoots) SplitIntoShares() NamespacedShares {
 
 func (msgs Messages) SplitIntoShares() NamespacedShares {
 	shares := make([]NamespacedShare, 0)
+	msgs.sortMessages()
 	for _, m := range msgs.MessagesList {
 		rawData, err := m.MarshalDelimited()
 		if err != nil {
 			panic(fmt.Sprintf("app accepted a Message that can not be encoded %#v", m))
 		}
-		shares = appendToShares(shares, m.NamespaceID, rawData)
+		shares = AppendToShares(shares, m.NamespaceID, rawData)
 	}
 	return shares
 }
 
+func (msgs *Messages) sortMessages() {
+	sort.Slice(msgs.MessagesList, func(i, j int) bool {
+		return bytes.Compare(msgs.MessagesList[i].NamespaceID, msgs.MessagesList[j].NamespaceID) < 0
+	})
+}
+
```

Comment:
must have forgot to pull this from #546, which is very important to stop the node from panicking when attempting to generate the extended data square. Otherwise, it's possible to to order message so that their namespaces are not in lexicographical order. tbh, idek how devnet is still alive